### ISSUE TITLE
Training OpenCLIP but GPU utilization is 0%

### ISSUE THREAD

**Issue by wsong1106** [mlfoundations/open_clip#681]
Hi, thanks for the great work. I am training OpenCLIP on my custom dataset, using the following command.

`python -m training.main --save-frequency 1  --zeroshot-frequency 1 --report-to tensorboard --train-data=/home/train.csv  --val-data=/home/val.csv -csv-img-key imgPath --csv-caption-key caption --warmup 10000 --batch-size=128 --lr=1e-5 --wd=0.1 --epochs=150 --workers=8 --model ViT-B-32
`
However, I can see my GPU memory has been allocated but the GPU utilization is always 0%. I wonder if you happen to know the reason and how I can resolve this. Many thanks.
![image](https://github.com/mlfoundations/open_clip/assets/33387501/1ecd4148-a644-49c7-90b8-bbae8a34507b)

**Comment by mitchellnw** [comment]
Does it print logs? I worry it's hanging before starting to train.

**Comment by wsong1106** [comment]
Yes it actually prints logs starting with `2023-10-18,18:06:24 | INFO | Running with a single process. Device cuda:0.`

And it doesn't just hang there, it starts the training although it is very slow. Here are some logs during training:
`2023-10-18,18:08:07 | INFO | Train Epoch: 0 [  256/65368 (0%)] Data (t): 96.245 Batch (t): 99.933, 2.56173/s, 2.56173/s/gpu LR: 0.000000 Logit Scale: 14.286 Contrastive_loss: 5.6169 (5.6169) Loss: 5.6169 (5.6169)
2023-10-18,18:27:13 | INFO | Train Epoch: 0 [25856/65368 (40%)] Data (t): 11.252 Batch (t): 11.458, 1243.01/s, 1243.01/s/gpu LR: 0.000000 Logit Scale: 14.286 Contrastive_loss: 5.5663 (5.5916) Loss: 5.5663 (5.5916)
2023-10-18,18:47:48 | INFO | Train Epoch: 0 [51456/65368 (79%)] Data (t): 12.141 Batch (t): 12.351, 3.23682/s, 3.23682/s/gpu LR: 0.000000 Logit Scale: 14.286 Contrastive_loss: 5.5183 (5.5672) Loss: 5.5183 (5.5672)
2023-10-18,18:57:20 | INFO | Train Epoch: 0 [65280/65368 (100%)] Data (t): 10.379 Batch (t): 10.592, 1285.53/s, 1285.53/s/gpu LR: 0.000000 Logit Scale: 14.286 Contrastive_loss: 5.4508 (5.5381) Loss: 5.4508 (5.5381)
2023-10-18,18:58:57 | INFO | Eval Epoch: 1 [256 / 8171] Clip Loss: 5.490433`

**Comment by wsong1106** [comment]
Found that the training actually was conducted using GPU, and the issue was rooted in that the data I/O took much longer than the training process..
